services/api/mw2pdf/classes/MediaWikiSession.js

```diff
@@ -0,0 +1,56 @@
+import { v4 as uuidv4 } from 'uuid'
+import puppeteer from 'puppeteer'
+
+import {
+  makeLoginRequest,
+  makeLoginTokenRequest,
+  extractLoginToken,
+} from '../utils/mediaWikiApi.js'
+
+import {
+  parseCookies,
+  toCookieString,
+} from '../utils/cookies.js'
+
+export class MediaWikiSession {
+  constructor(settings) {
+    this.config = {
```

Comment:
The proper way to do defaults in object based params: https://stackoverflow.com/a/26578323/159522